processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -149,4 +155,171 @@ public void remove()
         }
     );
   }
+
+  public static Filter convertToCNFFromQueryContext(Query query, Filter filter)
+  {
+    if (filter == null) {
+      return null;
+    }
+    boolean useCNF = (Boolean) query.getContextValue(CTX_KEY_USE_FILTER_CNF, false);
```

Comment:
1. IMO defaulting to false is good for now, some more performance testing work is needed to determine if true is a good default.
2. use `query.getContextBoolean` for booleans.